### PR TITLE
fix: clear selection after action

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[WEB-1679]

## Test Plan
- Check if selection will be 0 after action

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-1679]: https://hpe-aiatscale.atlassian.net/browse/WEB-1679?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/pages/F_ExpList/glide-table/TableActionBar.tsx [hunk 1]
```diff
@@ -155,7 +158,10 @@ const TableActionBar: React.FC<Props> = ({
   }, [experiments, project]);
 
   const selectedExperiments = useMemo(
-    () => Array.from(selectedExperimentIds).map((id) => experimentMap[id]),
+    () =>
+      Array.from(selectedExperimentIds).flatMap((id) =>
+        id in experimentMap ? [experimentMap[id]] : [],
+      ),
```
Comment (keita-determined) [inline webui/react/src/pages/F_ExpList/glide-table/TableActionBar.tsx:164]: in case `experimentMap` does not have `id`

**Review by ashtonG — CHANGES_REQUESTED** [review]
can we update the selection in [the handleActionSuccess callback](https://github.com/determined-ai/determined/blob/main/webui/react/src/pages/F_ExpList/F_ExperimentList.tsx#L420-L478) instead? that makes it so the table bar isn't responsible for the table selection.

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx [hunk 2]
```diff
@@ -575,6 +513,72 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
...
+        case ExperimentAction.DownloadCode:
+        case ExperimentAction.Edit:
+        case ExperimentAction.Fork:
+        case ExperimentAction.HyperparameterSearch:
+        case ExperimentAction.OpenTensorBoard:
+        case ExperimentAction.SwitchPin:
+        case ExperimentAction.ViewLogs:
+          break;
+      }
+      handleSelectionChange('remove-all', [0, selectedExperimentIds.size]);
```
Comment (keita-determined) [inline webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:572]: this line `handleSelectionChange('remove-all', [0, selectedExperimentIds.size]);` is the only change in this function and move this func to resolve `Block-scoped variable 'handleSelectionChange' used before its declaration` error

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx [hunk 3]
```diff
@@ -575,6 +513,72 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
...
+        // Exhaustive cases to ignore.
+        case ExperimentAction.CompareTrials:
+        case ExperimentAction.ContinueTrial:
+        case ExperimentAction.DownloadCode:
+        case ExperimentAction.Edit:
+        case ExperimentAction.Fork:
+        case ExperimentAction.HyperparameterSearch:
+        case ExperimentAction.OpenTensorBoard:
+        case ExperimentAction.SwitchPin:
+        case ExperimentAction.ViewLogs:
```
Comment (keita-determined) [inline webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:569]: should this be replaced with `default` clause?

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx [hunk 4]
```diff
@@ -62,6 +62,12 @@ interface Props {
   project: Project;
 }
 
+type selectionType = 'add' | 'add-all' | 'remove' | 'remove-all' | 'set';
```
Comment (hkang1) [inline webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:65]: **nit:** make the type uppercase `SelectionType` to be consistent

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx (hunk 3 above)
Comment (hkang1) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:569]: is typescript ok with just `default`?

**Review by hkang1 — COMMENTED** [review]
LGTM! minor comments

**Review by hkang1 — APPROVED** [review]
oops meant to approve

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx (hunk 3 above)
Comment (keita-determined) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:569]: it worked with it

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx (hunk 4 above)
Comment (keita-determined) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.tsx]: fixed

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx (hunk 3 above)
Comment (keita-determined) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.tsx]: replaced it with `default`

**Review by ashtonG — APPROVED** [review]

**Pull request merged**